provider/core/src/response.rs

```diff
@@ -127,36 +127,36 @@ where
 
 /// A wrapper type that wraps either an [`Rc`](alloc::rc::Rc) or an
 /// [`Arc`](alloc::sync::Arc), depending on the "sync" feature. Create
-/// this from a `&[u8]`.
+/// this from a `&[T]`.
 #[derive(Clone)]
-pub struct RcWrap(
-    #[cfg(not(feature = "sync"))] alloc::rc::Rc<[u8]>,
-    #[cfg(feature = "sync")] alloc::sync::Arc<[u8]>,
+pub struct RcWrap<T>(
+    #[cfg(not(feature = "sync"))] alloc::rc::Rc<[T]>,
+    #[cfg(feature = "sync")] alloc::sync::Arc<[T]>,
 );
 
-impl core::ops::Deref for RcWrap {
-    type Target = [u8];
+impl<T> core::ops::Deref for RcWrap<T> {
+    type Target = [T];
     fn deref(&self) -> &Self::Target {
         self.0.deref()
     }
 }
 
 // Safe because both Rc and Arc are CloneableCart
-unsafe impl CloneableCart for RcWrap {}
+unsafe impl<T: Clone> CloneableCart for RcWrap<T> {}
```

Comment:
Nit: I think you don't need the `T: Clone` bound on this impl

Reply:
Yeah, Rc is unconditionally Clone

You may need `T: ?Sized` for `[u8]` to work here

Reply:
This is done. Implemented custom Clone for the reason I described above.